pkg/policy/mapstate.go

```diff
@@ -345,25 +392,47 @@ func (keys MapState) denyPreferredInsertWithChanges(newKey Key, newEntry MapStat
 				}
 			}
 		default:
-			// Do not insert 'newKey' if the map state already denies traffic
+			// We do not insert 'newKey' if the map state already denies traffic
 			// which is a superset of (or equal to) 'newKey'
```

Comment:
nit: With the current code, I think that if we attempt to insert a new L3 deny that is more specific than an existing L3 deny (eg insert a deny 10.0.0.0/8 when 0.0.0.0/0 is already denied), this will result in two entries being inserted, one for each deny.  I think that this is ~OK, but it's subtly different from what this (existing) comment is describing, which is that in such a case we wouldn't insert the deny 10.0.0.0/8.

In order for the code to match the comment, we would probably need to do a check here also for whether there is any superset deny of `newKey` in `keys` and skip in that case also. But that said, I'm not sure we actually need the implementation to do that. There's pluses and minuses to this - more iteration while handling inserts, but fewer map entries in this corner case.

One alternative is something like:
```suggestion
			// We do not insert 'newKey' if the map state already denies traffic
			// which is a superset of (or equal to) 'newKey'.
			// Caveat: In the case of overlapping CIDRs, multiple denies may be
			// inserted. We could optimize this further to minimize the set of
			// map entries, but the current implementation is functionally correct.
```